File: service/src/main/java/org/vsegda/service/AlertService.java
package org.vsegda.service;

import org.vsegda.data.DataStream;

import javax.mail.Message;
import javax.mail.MessagingException;
import javax.mail.Session;
import javax.mail.Transport;
import javax.mail.internet.InternetAddress;
import javax.mail.internet.MimeMessage;
import java.io.IOException;
import java.util.Properties;

/**
 * @author Roman Elizarov
 */
public class AlertService {
    public static void sendAlertEmail(String code, String text) throws IOException {
        Session session = Session.getDefaultInstance(new Properties(), null);
        String subject = "ALERT: " + text;
        String body = text;
        if (code != null) {
            DataStream stream = DataStreamService.INSTANCE.resolveDataStreamByCode(code);
            subject += ": " + (stream != null ? stream.getNameOrCode() : code);
            body += "\nhttp://apps.vsegda.org/dataPlot?id=" + code;
        }
        try {
            Message msg = new MimeMessage(session);
            msg.setFrom(new InternetAddress("vsdacha@gmail.com", "Dacha"));
            msg.addRecipient(Message.RecipientType.TO, new InternetAddress("elizarov@gmail.com", "Roman Elizarov"));
            msg.setSubject(subject);
            msg.setText(body);
            Transport.send(msg);
        } catch (MessagingException e) {
            throw new IOException(e);
        }
    }
}

File: service/src/main/java/org/vsegda/service/AlertService.kt
package org.vsegda.service

import org.vsegda.util.*
import java.io.*
import java.util.*
import javax.mail.*
import javax.mail.internet.*

object AlertService {
    fun sendAlertEmail(code: String?, text: String) {
        val session = Session.getDefaultInstance(Properties(), null)
        var subject = "ALERT: " + text
        var body = text
        if (code != null) {
            val stream = DataStreamService.resolveDataStreamByCode(code)
            subject += ": " + (stream?.nameOrCode ?: code)
            body += "\nhttp://apps.vsegda.org/dataPlot?id=" + code
        }
        val from = InternetAddress("vsdacha@gmail.com", "Dacha")
        val to = InternetAddress("elizarov@gmail.com", "Roman Elizarov")
        log.info("Sending email from=$from to=$to subject=$subject")
        Transport.send(MimeMessage(session).apply {
            setFrom(from)
            addRecipient(Message.RecipientType.TO, to)
            setSubject(subject)
            setText(body)
        })
    }
}

